src/main/object-uploader.js

```diff
@@ -207,7 +208,43 @@ export default class ObjectUploader extends Transform {
   }
 
   _flush(callback) {
+   if (this.emptyStream) {
```

Comment:
Test works fine, I was able to successfully upload an empty object. However, is there any particular reason why we have this code inside the `flush` function? If it isn't crucial for this code to be written inside the `flush` function, I think it might be better to handle this case inside the `transform` function to avoid repetition of code